megamek/src/megamek/common/alphaStrike/conversion/ASLocationMapper.java

```diff
@@ -78,7 +78,7 @@ public static double damageLocationMultiplier(Entity en, int loc, Mounted mount)
             // Don't count squad support weapons, these are handled separately
             return ((mount.getLocation() <= 1) && !mount.isSquadSupportWeapon()) ? 1 : 0;
         } else if (en instanceof Infantry) {
-            return (loc == mount.getLocation()) ? 1 : 0;
+            return (loc == mount.getLocation()) ^ ((Infantry)en).hasFieldWeapon() ? 1 : 0;
```

Comment:
```suggestion
            return (loc == mount.getLocation()) ^ ((Infantry) en).hasFieldWeapon() ? 1 : 0;
```

Reply:
Also, is this supposed to be an XOR?

Reply:
Yes, that was intentional. But I've rewritten it anyway to be more digestible. It should only count field weapons when it has field weapons and when it hasn't, it can count all its weapons.